codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/StructureGenerator.kt

```diff
@@ -113,6 +120,36 @@ class StructureGenerator(
         }
     }
 
+    private fun renderStructureImpl() {
+        if (members.isEmpty()) {
+            return
+        }
+        writer.rustBlock("impl $name") {
+            // Render field accessor methods
+            forEachMember { member, memberName, memberSymbol ->
```

Comment:
this isn't quite right. It will make an empty impl block for errors. Could add an extension function to produce a list of members we're actually going to render?